inc/components/class-component.php

```diff
@@ -543,13 +543,21 @@ public function append_child( $child ): self {
 	 * @return self
 	 */
 	private function hydrate_config(): self {
+		/**
+		 * Filter the config values before the config callback fires.
+		 *
+		 * @param array  $config Config for this component.
+		 * @param string $name   Name of this component.
+		 */
+		$new_config = apply_filters( 'wp_irving_component_config', $this->get_config(), $this->get_name() );
```

Comment:
Let's rename this something like `wp_irving_component_config_pre_hydration` or `wp_irving_component_config_data` so we can save `wp_irving_component_config` as a hook that fires after hydration—consistent with the `wp_irving_component_children` hook.